src/index.ts

```diff
@@ -209,17 +202,26 @@ export const derive = async function (privateKeyA: Buffer, publicKeyB: Buffer):
   if (publicKeyB.length === 33) {
     assert(publicKeyB[0] === 2 || publicKeyB[0] === 3, "Bad public key");
   }
-  const keyA = ec.keyFromPrivate(privateKeyA);
-  const keyB = ec.keyFromPublic(publicKeyB);
-  const Px = keyA.derive(keyB.getPublic()); // BN instance
-  return Buffer.from(Px.toArray());
+
+  // unpad to match previous implementation
+  // elliptic return BN and we return Buffer(BN.toArray())
+  // match by unpadding
+  const sharedSecret = secp256k1.getSharedSecret(privateKeyA, publicKeyB);
+  const Px = sharedSecret.subarray(sharedSecret.length - 32);
+
+  let i = 0;
+  while (i < Px.length && Px[i] === 0) {
+    i++;
+  }
+
+  return Px.subarray(i);
```

Comment:
can be simpliefied
```suggestion
  // All points on secp256k1 have x!=0, so this will always find an index.
  const i = Px.findIndex((byte) => byte !== 0);
  return Px.subarray(i);
```